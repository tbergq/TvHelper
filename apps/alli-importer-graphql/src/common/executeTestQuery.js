// @flow

import { graphql } from 'graphql';

import schema from '../Schema';

const createContext = () => ({});

export default function executeTestQuery(
  query: string,
  variables: ?{ ... },
  request: ?{ ... } = {},
): Promise<{ ... }> {
  // $FlowExpectedError: Ok for testing purposes
  return graphql(schema, query, null, createContext(request), variables);
}
