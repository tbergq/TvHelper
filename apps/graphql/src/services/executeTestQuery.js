// @flow

import { graphql } from 'graphql';

import schema from '../Schema';
import createContext from './createGraphqlContext';

export default function executeTestQuery(
  query: string,
  variables: ?{ ... },
  request: ?{ ... } = {},
): Promise<{ ... }> {
  // $FlowExpectedError: Ok for testing purposes
  return graphql(schema, query, null, createContext(request), variables);
}
