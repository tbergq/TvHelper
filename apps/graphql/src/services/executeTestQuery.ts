import { graphql } from 'graphql';

import schema from '../Schema';
import createContext from './createGraphqlContext';

export default function executeTestQuery(
  query: string,
  variables: {} | null | undefined,
  request: {} | null | undefined = {},
): Promise<{}> {
  // @ts-ignore: Ok for testing purposes
  return graphql(schema, query, null, createContext(request), variables);
}
