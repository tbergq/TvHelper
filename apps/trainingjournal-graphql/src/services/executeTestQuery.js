// @flow

import { graphql } from 'graphql';

import createContext from '../createContext';
import schema from '../Schema';

export function getOperationName(query: string): ?string {
  const match = query.match(/(?:query|mutation)\s(?<queryName>[^\s(]*)/);
  const name = match?.groups?.queryName;
  return name;
}

export default function executeTestQuery(
  query: string,
  variables: ?{ ... },
  req: ?{ ... } = { user: null },
): Promise<{ ... }> {
  // $FlowExpectedError
  return graphql(schema, query, null, createContext({ req }), variables);
}
