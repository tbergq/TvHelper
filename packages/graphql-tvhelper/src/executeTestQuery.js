// @flow

import { graphql as originalGraphQL, GraphQLSchema, GraphQLObjectType } from 'graphql';

import SearchTvShow from './search/queries/SearchTvShow';

const RootQuery = new GraphQLObjectType({
  name: 'RootQuery',
  description: 'Root Query',
  fields: {
    searchTvShow: SearchTvShow,
  },
});

const schema = new GraphQLSchema({
  query: RootQuery,
});

/**
 * Executes GraphQL query using our schema. This function should
 * be used only during testing because it uses faked context.
 */
export default function executeTestQuery(
  query: string,
  variables: ?{ ... },
  context: ?{ ... },
): Promise<Object> {
  return originalGraphQL(schema, query, null, context, variables);
}
