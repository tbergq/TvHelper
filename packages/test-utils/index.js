// @flow

export { default as withMongoDb } from './src/useMongoDb';
