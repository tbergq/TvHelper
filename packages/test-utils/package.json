{
  "name": "@tbergq/test-utils",
  "version": "0.0.1",
  "private": true,
  "main": "index.js",
  "dependencies": {
    "mongodb-memory-server": "^5.2.0",
    "mongoose": "^5.6.9"
  }
}
