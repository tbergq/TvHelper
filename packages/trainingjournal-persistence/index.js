// @flow strict

import Exercise from './src/dataObjects/Exercise';

// Repositories
export { default as UserRepository } from './src/repositories/UserRepository';
export { default as ExerciseRepository } from './src/repositories/ExerciseRepository';
export { default as ProgramRepository } from './src/repositories/ProgramRepository';

// Connection
export { default as trainingJournalConnection } from './src/connection';

// dataObject

export type ExerciseType = typeof Exercise;
