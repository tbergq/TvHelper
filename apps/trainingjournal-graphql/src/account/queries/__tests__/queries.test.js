// @flow

import { UserRepository, ExerciseRepository } from '@tbergq/trainingjournal-persistence';
import { generateTestsFromFixtures } from '@adeira/test-utils';
import path from 'path';
import Mockdate from 'mockdate';

import executeTestQuery from '../../../services/executeTestQuery';

Mockdate.set(new Date(2020, 0, 1, 0, 0, 0, 0));

let userId;
beforeEach(async () => {
  // $FlowExpectedError[incompatible-call]
  const { id } = await UserRepository.createUser({
    username: 'tita',
    password: 'password',
    email: 'tita@bebita.com',
  });
  userId = id;
  await ExerciseRepository.createExercise({
    userId: id,
    name: 'Squat',
  });
  await ExerciseRepository.createExercise({
    userId: id,
    name: 'Bench press',
  });
});

generateTestsFromFixtures(path.join(__dirname, '__fixtures__'), (input) => {
  return executeTestQuery(input, null, { user: { id: userId } });
});
