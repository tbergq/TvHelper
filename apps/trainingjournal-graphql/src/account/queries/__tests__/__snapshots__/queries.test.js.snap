// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches expected output: exercises.graphql 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
query exercises {
   viewer {
    ... on User {
      exercises {
        edges {
          node {
            id
            name
          }
        }
      }
    }
  }
}

~~~~~~~~~~ OUTPUT ~~~~~~~~~~
{
  "data": {
    "viewer": {
      "exercises": {
        "edges": [
          {
            "node": {
              "id": "RXhlcmNpc2U6MQ",
              "name": "Squat"
            }
          },
          {
            "node": {
              "id": "RXhlcmNpc2U6Mg",
              "name": "Bench press"
            }
          }
        ]
      }
    }
  }
}
`;
